apps/frontend/src/components/UserProfileCard.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { Box, Flex, HStack, IconButton, Image, Text } from '@chakra-ui/react';
+
+// If more user data is needed, it can be added here
+type UserData = {
+  name: string,
+  image?: string
+}
+
+type UserProfileIcon = {
```

Comment:
nitpick: This name is a bit confusing as "icons" are non-interactive elements, i.e. they are the visual component of a button (which is evidenced by `icon` being a property below). I'd suggest simply calling it `CardButton` or something (as they're not specific to user profiles).

Also should they not also have a onclick function property?

Reply:
Good point. Changed name to `CardButton`.

Reply:
Looks like you actually changed it to `UserButton`. That makes it's purpose (a button) a little more clear, but it may be inaccurate (or too specific) - is it a button than displays a user? If the button can be used for various purposes (I think you said it might be used in Item cards too), but always within a card, then `CardButton` would be better.

Also remember to look for any references to a modified class/type and update accordingly. i.e. in `TestPage.tsx` you still call the list of buttons `icons`.

Reply:
Yes, my fault. Working too fast, thinking too slow. Will update this and adjust accordingly. Thanks for pointing that out.

Reply:
Also, as mentioned since they're buttons they should have a function associated with them.

Reply:
Added a general onClick event. We can pass a specific implementation when we create the buttons and pass them to the UserProfileCard.

<img width="588" alt="Screenshot 2024-03-08 at 11 22 58" src="https://github.com/Geek-Collectors-Network/geek-collectors-network/assets/97256979/0cf50063-7524-474b-a6d7-519c971cccbc">

Reply:
Updated.